### PR TITLE
megasync: 4.9.0.0 -> 4.9.1.0

### LINKED ISSUE #290949: Megasync and megacmd refuse to build, because package ‘freeimage-unstable-2021-11-01’  is marked as insecure
### Describe the bug
Megasync and megacmd won't build, because one of the packages they depend on, ‘freeimage-unstable-2021-11-01’  is marked as insecure. The error reads: 

error: Package ‘freeimage-unstable-2021-11-01’ in /nix/store/gzf4zwcakda1nykn6h0avh45xhjhvsz4-source/pkgs/development/libraries/freeimage/default.nix:72 is marked as insecure, refusing to evaluate.

### Steps To Reproduce
1. Try building a system which includes the packages `megasync` or `megacmd`
2. Notice that you get an error `error: Package ‘freeimage-unstable-2021-11-01’ in /nix/store/gzf4zwcakda1nykn6h0avh45xhjhvsz4-source/pkgs/development/libraries/freeimage/default.nix:72 is marked as insecure, refusing to evaluate.`

### Expected behavior
I expect to be able to build my system which contains `megasync` and `megacmd`

### Notify maintainers

@michojel 
@alkeryn 
@Synthetica9 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.5, NixOS, 24.05 (Uakari), 24.05.20240218.b98a4e1`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(jon): `"nixos"`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by michojel**
and removed freeimage dependency

Resolves: #290949

Unfortunately, non of the newer upstream releases worked for me. Either they crash or get [stuck at fetching the file list](https://github.com/meganz/MEGAsync/issues/839)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/misc/megasync/default.nix [hunk 1]
```diff
@@ -72,7 +70,6 @@ mkDerivation rec {
     ./noinstall-distro-version.patch
     # megasync target is not part of the install rule thanks to a commented block
     ./install-megasync.patch
-    ./ffmpeg_44.patch
```
Comment (ludat) [inline pkgs/applications/misc/megasync/default.nix:75]: shouldn't we delete the patch file as well if we are not using it?

File: pkgs/applications/misc/megasync/default.nix [hunk 2]
```diff
@@ -98,7 +95,7 @@ mkDerivation rec {
     "--with-cryptopp"
     "--with-curl"
     "--with-ffmpeg"
-    "--with-freeimage"
+    "--without-freeimage"
```
Comment (ludat) [inline pkgs/applications/misc/megasync/default.nix:98]: Is this necessary? seems like disabling freeimage has to be done in too many places :thinking:

File: pkgs/applications/misc/megasync/default.nix (hunk 1 above)
Comment (michojel) [reply pkgs/applications/misc/megasync/default.nix:75]: Thanks for reminding me. Removed.

File: pkgs/applications/misc/megasync/default.nix (hunk 2 above)
Comment (michojel) [reply pkgs/applications/misc/megasync/default.nix:98]: It looks to be necessary. Without the line the compilation fails like this:

```
$ nix build --nix-path nixpkgs=/home/michojel/.cache/nixpkgs-review/rev-b3b106b7a6d0d9d1300b54a4fe3ff4f98179afb1/nixpkgs nixpkgs-overlays=/tmp/tmp9zpa397q --extra-experimental-features nix-command no-url-literals --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed -f /home/michojel/.cache/nixpkgs-review/rev-b3b106b7a6d0d9d1300b54a4fe3ff4f98179afb1/build.nix
error: builder for '/nix/store/mdkzammq8z6m792i1al65kxpyqc95ymg-megasync-4.9.1.0.drv' failed with exit code 1;
       last 10 log lines:
       > checking for SSL_new in -lssl... yes
       > checking for c-ares... checking for ares.h... yes
       > checking for ares_library_init in -lcares... yes
       > checking for curl/curl.h... yes
       > checking for main in -lcurl... yes
       > checking whether cURL is built with OpenSSL support... yes
       > checking if building example applications... no
       > checking for FreeImage... --with-freeimage not specified
       > checking for FreeImage.h... no
       > configure: error: FreeImage.h header not found or not usable
       For full logs, run 'nix log /nix/store/mdkzammq8z6m792i1al65kxpyqc95ymg-megasync-4.9.1.0.drv'.
error: 1 dependencies of derivation '/nix/store/xgazyqgns7v6dl9ywy7fz83nxvfnwa04-review-shell.drv' failed to build
1 package failed to build:
megasync
```

**Review by ludat — APPROVED** [review]
tested it out on my machine and it seems to work

**Pull request merged**